docs/miscellaneous.rst

```diff
@@ -58,6 +61,51 @@ So for the following contract snippet::
 
 The position of ``data[4][9].b`` is at ``keccak256(uint256(9) . keccak256(uint256(4) . uint256(1))) + 1``.
 
+Adding elements to or removing them from a dynamically-sized array will increment or decrement the number
+of elements which is stored in the slot. For arrays that are not of type ``bytes`` or ``string`` this is
+straight-forward, since data is stored in a different slot.
+
+
+``bytes`` and ``string``
+------------------------
+
+``bytes`` and ``string`` store their data in the same slot where also the length is stored if they are short. In
+particular: If the data is at most ``31`` bytes long, it is stored in the higher-order bytes (left aligned) and the
+lowest-order byte stores ``length * 2``. If it is longer, the main slot stores ``length * 2 + 1`` and the data is
+stored as usual in ``keccak256(slot)``.
+
+So for the following contract snippet::
+
+    pragma solidity ^0.4.0;
+
+    contract C {
+      bytes data;
+      function f() {
+        data.push(3);
+        data.push(5);
+        data.push(7);
+      }
+    }
+
+The storage layout will look like this (slot is 32 bytes, most-significant byte is on the left)::
+
+    03 05 07 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 03
+
+Calling ``push()`` or ``pop()`` will just add ``2`` to or subtract ``2`` from the length (lowest-order byte)
```

Comment:
It might be confusing to talk about how to change the size, perhaps we should only talk about how to encode and decode data.